bids/layout/models.py

```diff
@@ -32,35 +28,60 @@ def __init_subclass__(cls, extension, **kwargs):
             BIDSFile._ext_registry[ext] = cls
 
     @classmethod
-    def from_path(cls, path):
-        path = Path(path)
+    def from_filename(cls, filename):
+        path = Path(filename)
         for ext, subclass in cls._ext_registry.items():
             if path.name.endswith(ext):
                 cls = subclass
                 break
-        return cls(path)
+        return cls(filename=path)
 
-    def __init__(self, filename, root=None):
-        self.path = str(filename)
-        self.filename = self._path.name
-        self.dirname = str(self._path.parent)
-        self.is_dir = not self.filename
-        self._root = root
+    @classmethod
+    def from_artifact(cls, artifact):
+        """ Load from ANCPBids Artifact """
+        return cls(artifact=artifact)
+
+    def __init__(self, artifact=None, filename=None):
+        if artifact is not None:
+            self.artifact = artifact
+        elif filename is not None:
+            self.artifact = Artifact(**parse_bids_name(filename))
```

Comment:
the user must provide a pointer to the used schema or it must be inferred from some context... do we really need an ancpBIDS Artifact to be constructed in BIDSFile?

Reply:
the constructed Artifact is not attached to a Dataset graph... I think it would be better to integrate constructing BIDSFile instances from an existing BIDSLayout or just deprecate BIDSFile

Reply:
not necessarily, but the most common use case for this now is simply to mirror the `BIDSFile` API with results from ancpbids. I suppose I could also just take apart the `Artifact` when that is given as an input, and only use `parse_bids_name` when given a filename as input.

yes, I need to adjust this to point to a specific schema version. The only use case for creating a `BIDSFile` from filename (rather than from ancpbids query results), is for writing files, so I suppose defaulting to the latest version of the schema unless otherwise explicitly stated is reasonable.

Reply:
there is already #1 to return BIDSFiles instead of Artifacts

Reply:
> so, instead of BIDSLayout.get() returning ancpBIDS Artifact object, return BIDSFiles which just wrap Artifacts

yes, I agree, this is intended to basically be a light wrapper for `Artifact` objects, with potentially a deprecation cycle for `BIDSFile` in the future. 

Do you mean specifically to inherit `Artifact` for `BIDSFile`, that would be easier potentially

Reply:
inheritance over composition (wrapping) is also a valid and in this case good approach

Reply:
I think inheritance might work in the future if we align our terminology (i.e. `path` vs `filename` vs `name` etc...) and API, so in the meantime I prefer to wrap

Reply:
Given that, can I clarify that your suggestion is that `BIDSFile` should only be a wrapper for `Artifact` but _not_ also a constructor of `Artifacts`?

i.e.:
```
    def __init__(self, artifact):
       self._artifact = artifact
```

and the rare scenarios where we might want to construct a `BIDSFile` in pybids in a different context (i.e. to mimic legacy file writing API), we can construct outside of this wrapper?